src/middleware/tests/auth.middleware.test.ts

```diff
@@ -0,0 +1,149 @@
+import { AccessToken, RefreshToken } from 'devu-shared-modules'
+
+import environment from '../../environment'
+
+import { isAuthorized, isValidRefreshToken, isRefreshNearingExpiration } from '../auth.middleware'
+
+import AuthService from '../../services/auth.service'
+
+import Testing from '../../utils/testing.utils'
+import { Unauthorized } from '../../utils/apiResponse.utils'
+
+// Testing Globals
+let req: any
+let res: any
+let next: any
+
+let expectedAccessToken: AccessToken = { userId: 1, email: '' }
+let expectedRefreshToken: RefreshToken = { userId: 1, isRefreshToken: true }
+// let expectedError = new Error('Some error')
+
+describe('AuthMiddleware', () => {
+  beforeEach(() => {
+    req = Testing.fakeRequest()
+    res = Testing.fakeResponse()
+    next = Testing.fakeNext()
+
+    req.headers.authorization = 'Bearer heehee.nottoken'
+  })
+
+  describe('isAuthenticated', () => {
+    describe('Authorization Successful', () => {
+      beforeEach(async () => {
+        AuthService.validateJwt = jest.fn().mockImplementation(<AccessToken>() => expectedAccessToken)
+
+        await isAuthorized(req, res, next)
+      })
+
+      test('Expect access token to be added to req', () => expect(req.currentUser).toEqual(expectedAccessToken))
+
+      test('Expect next to be called on successful token validation', async () => expect(next).toHaveBeenCalled())
+      test('Next is called', () => expect(next).toHaveBeenCalledTimes(1))
+    })
+
+    describe('Authorization Failure', () => {
+      test('Missing authroization header', async () => {
+        delete req.headers.authorization
+        await isAuthorized(req, res, next)
+
+        expect(res.status).toBeCalledWith(401)
+        expect(next).toHaveBeenCalledTimes(0)
+      })
+
+      test('Missing Bearer', async () => {
+        req.headers.authorization = 'NotBearer and.a.fake.token'
+        await isAuthorized(req, res, next)
+
+        expect(res.status).toBeCalledWith(401)
+        expect(next).toHaveBeenCalledTimes(0)
+      })
+
+      test('Missing Token', async () => {
+        req.headers.authorization = 'Bearer' // missing token
+        await isAuthorized(req, res, next)
+
+        expect(res.status).toBeCalledWith(401)
+        expect(next).toHaveBeenCalledTimes(0)
+      })
+      test('Invalid token', async () => {
+        AuthService.validateJwt = jest.fn().mockImplementation(<AccessToken>() => null)
+        await isAuthorized(req, res, next)
+
+        expect(res.status).toHaveBeenCalledWith(401)
+        expect(next).toHaveBeenCalledTimes(0)
+      })
+    })
+  })
+
+  describe('isValidRefresToken', () => {
```

Comment:
```suggestion
  describe('isValidRefreshToken', () => {
```

Reply:
This suite also isnt validating that a refresh token is in fact a refresh token via `isRefreshToken` and bailing accordingly if that is not the case.